src/walletSdk/Anchor/Sep6.ts

```diff
@@ -177,4 +190,104 @@ export class Sep6 {
       throw e;
     }
   }
+
+  /**
+   * Creates a new instance of the Watcher class, to watch sep6 transactions.
+   * @returns {Watcher} A new Watcher instance.
+   */
+  watcher(): Watcher {
+    return new Watcher(this.anchor, WatcherSepType.SEP6);
+  }
+
+  /**
+   * Get account's sep6 transactions specified by asset and other params.
+   * @param {GetTransactionsParams} params - The Get Transactions params.
+   * @param {AuthToken} params.authToken - The authentication token for the account authenticated with the anchor.
+   * @param {string} params.assetCode - The target asset to query for.
+   * @param {string} params.account - The stellar account public key involved in the transactions. If the service requires SEP-10
+   * authentication, this parameter must match the authenticated account.
+   * @param {string} [params.noOlderThan] - The response should contain transactions starting on or after this date & time.
+   * @param {string} [params.limit] - The response should contain at most 'limit' transactions.
+   * @param {string} [params.kind] - The kind of transaction that is desired. E.g.: 'deposit', 'withdrawal', 'depo
+   * -exchange', 'withdrawal-exchange'.
+   * @param {string} [params.pagingId] - The response should contain transactions starting prior to this ID (exclusive).
+   * @param {string} [params.lang] - The desired language (localization), it can also accept locale in the format 'en-US'.
+   * @returns {Promise<Sep6Transaction[]>} A list of transactions as requested by the client, sorted in time-descending order.
+   * @throws {InvalidTransactionsResponseError} Anchor returns an invalid response.
+   * @throws {ServerRequestFailedError} If server request fails.
+   */
+  async getTransactionsForAsset({
+    authToken,
+    assetCode,
+    account,
```

Comment:
Why do we need an account here? Can't we get it from the auth token? I think in real implementations there are no use cases where account != SEP-10 account

Reply:
Sep 6 requires it. If you don't give an account that matches the auth token account it gives a 403. 

I'm not with my laptop right now but I can link to it on the sep 6 spec tonight

Reply:
I'm suggesting to omit it in the method parameter, and instead always pass account from the auth token to the anchor. (that's how anchor platform and polaris work, as they require authentication in /transaction endpoint, therefore account must always match authenticated account)

Reply:
ahh I see. Ya I agree that's better. We have this ticket for tying the account to the auth token (right now it's just a string) https://stellarorg.atlassian.net/browse/WAL-1152?atlOrigin=eyJpIjoiYjMxYWY2NDRkY2IyNGE4MGE4MjlkNWM3NDFmNzVlMDQiLCJwIjoiaiJ9

I'll make a note on the ticket to change this as well to always pull from the auth token